contracts/upgradeable_contracts/erc20_to_native/ForeignBridgeErcToNative.sol

```diff
@@ -171,7 +175,61 @@ contract ForeignBridgeErcToNative is BasicForeignBridge, ERC20Bridge, OtherSideB
         emit TokensSwapped(hdToken, fdToken, curHDTokenBalance);
     }
 
-    function relayTokens(address _from, address _receiver, uint256 _amount, address _token) external {
+    function setMinDaiTokenBalance(uint256 _minBalance) external onlyOwner {
+        uintStorage[MIN_DAI_TOKEN_BALANCE] = _minBalance;
+    }
+
+    /**
+    * @dev Evaluates edge DAI token balance, which has an impact on the invest amounts
+    * @return Value in DAI
+    */
+    function minDaiTokenBalance() public view returns (uint256) {
+        return uintStorage[MIN_DAI_TOKEN_BALANCE];
+    }
+
+    /**
+    * @dev Checks if DAI balance is high enough to be partially converted to Chai
+    * Twice limit is used in order to decrease frequency of convertDaiToChai calls,
+    * In case of high bridge utilization in DAI => xDAI direction,
+    * convertDaiToChai() will be called as soon as DAI balance reaches 2 * limit,
+    * limit DAI will be left as a buffer for future operations.
+    * @return true if convertDaiToChai() call is needed to be performed by the oracle
+    */
+    function isDaiNeedsToBeInvested() public view returns (bool) {
+        // chai token needs to be initialized, DAI balance should be at least twice greater than minDaiTokenBalance
+        return address(chaiToken()) != address(0) && tokenBalance(erc20token()) > 2 * minDaiTokenBalance();
+    }
+
+    /**
+    * @dev Converts all DAI into Chai tokens, keeping minDaiTokenBalance() DAI as a buffer
+    */
+    function convertDaiToChai() public {
+        _convertDaiToChai(tokenBalance(erc20token()).sub(minDaiTokenBalance()));
+    }
+
+    /**
```

Comment:
Consider to remove `convertDaiToChai(amount)` and `convertChaiToDai(amount)` to reduce the power of the owner. It is assumed that the owner is responsible only for changing the parameters of the bridge. It also will allows to safe some gas during the contract deployment.